### PR TITLE
Allow Updating package.lock file through Actions

### PR DESCRIPTION
**Pull request by garrettmflynn**
An attempt to fix issues when we push an M1 / M2 package.lock file (#598, #597)

### CODE REVIEW

File: .github/workflows/update-package.yml [hunk 1]
```diff
@@ -0,0 +1,25 @@
+name: Update package.lock
+
+on:
+    workflow_dispatch:
+
+jobs:
+  deploy:
+    runs-on: windows-latest
```
Comment (CodyCBakerPhD) [inline .github/workflows/update-package.yml:8]: But you said this is specifically for non-M1 mac, right?

File: .github/workflows/update-package.yml [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
```
Comment (CodyCBakerPhD) [inline .github/workflows/update-package.yml:8]:
```suggestion
    runs-on: macos-latest
```

**Comment by CodyCBakerPhD** [comment]
Also

> An attempt to fix issues when we push an M1 / M2 package.lock file (https://github.com/NeurodataWithoutBorders/nwb-guide/issues/598, https://github.com/NeurodataWithoutBorders/nwb-guide/issues/597)

What does this PR have to do with those previous problems? We're not running into a currently problem with building the app are we?

The first PR had to do with M1 specifically (our current issue is with non-M1/M2) and the second PR was the conda environment and is unaffected by the `npm` side

**Comment by garrettmflynn** [comment]
Both of those issues had aspects that were fixed by https://github.com/NeurodataWithoutBorders/nwb-guide/pull/605, which would be reversed by pushing a compromised package-lock.json from (at least) my M2

**Comment by CodyCBakerPhD** [comment]
Hah, IDK - even `npm install` might be running into issues now: https://github.com/NeurodataWithoutBorders/nwb-guide/actions/runs/8287280865/job/22679223021?pr=673

File: .github/workflows/update-package.yml [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+jobs:
+  deploy:
+    runs-on: macos-latest
+    steps:
+      - uses: actions/checkout@v4
+      - name: Checkout New Branch
+        run: git checkout -b update_package_lock
+      - name: Update package.lock file
+        run: npm install
```
Comment (CodyCBakerPhD) [inline .github/workflows/update-package.yml:15]:
```suggestion
        run: npm install --verbose
```

**Comment by CodyCBakerPhD** [comment]
Added verbosity: seems to stall at this line: https://github.com/NeurodataWithoutBorders/nwb-guide/actions/runs/8287381363/job/22679580477?pr=673#step:4:1865

**Comment by CodyCBakerPhD** [comment]
replaced by #678

**Pull request closed without merging**